test/src/module/common/walFilterTest.c

```diff
@@ -0,0 +1,1848 @@
+{
+    Buffer *filtred = bufNew(1024 * 1024);
+    Buffer *output = bufNew(outputSize);
+    ioBufferSizeSet(inputSize);
+
+    IoRead *read = ioBufferReadNew(wal);
+    ioFilterGroupAdd(ioReadFilterGroup(read), filter);
+    ioReadOpen(read);
+
+    while (!ioReadEof(read))
+    {
+        ioRead(read, output);
+        bufCat(filtred, output);
+        bufUsedZero(output);
+    }
+
+    ioReadClose(read);
+    bufFree(output);
+    ioFilterFree(filter);
+
+    return filtred;
+}
+
+#define buildWalP(wal, records, count, flags, ...) \
+    buildWal(wal, records, count, flags, (buildWalParam){VAR_PARAM_INIT, __VA_ARGS__})
+
+static __attribute__((unused)) void
+buildWal(Buffer *wal, XRecordInfo *records, size_t count, WalFlags flags, buildWalParam param)
+{
+    if (param.pageSize == 0)
+    {
+        param.pageSize = DEFAULT_GDPB_XLOG_PAGE_SIZE;
+    }
+
+    for (size_t i = 0; i < count; ++i)
+    {
+        XLogRecord *record = hrnGpdbCreateXRecord(records[i].rmid, records[i].info, records[i].body_size, records[i].body);
+        hrnGpdbWalInsertXRecordP(wal, record, NO_FLAGS, .walPageSize = param.pageSize);
+    }
+    if (!(flags & NO_SWITCH_WAL))
+    {
+        XLogRecord *record = hrnGpdbCreateXRecord(0, XLOG_SWITCH, 0, NULL);
+        hrnGpdbWalInsertXRecordP(wal, record, NO_FLAGS, .walPageSize = param.pageSize);
+        size_t to_write = param.pageSize - bufUsed(wal) % param.pageSize;
+        memset(bufRemainsPtr(wal), 0, to_write);
+        bufUsedInc(wal, to_write);
+    }
+}
+
+static void
+fillLastPage(Buffer *wal, PgPageSize pageSize)
+{
+    if (bufUsed(wal) % pageSize == 0)
+    {
+        return;
+    }
+    size_t to_write = pageSize - bufUsed(wal) % pageSize;
+    memset(bufRemainsPtr(wal), 0, to_write);
+    bufUsedInc(wal, to_write);
+}
+
+static void
+testGetRelfilenode(uint8_t rmid, uint8_t info, bool expect_not_skip)
+{
+    RelFileNode node = {1, 2, 3};
+
+    XLogRecord *record = hrnGpdbCreateXRecord(rmid, info, sizeof(node), &node);
+
+    const RelFileNode *node_result = getRelFileNodeGPDB6(record);
+    RelFileNode node_expect = {1, 2, 3};
+    TEST_RESULT_BOOL(node_result != NULL, expect_not_skip, "RelFileNode is different from expected");
+    if (expect_not_skip)
+        TEST_RESULT_BOOL(memcmp(&node_expect, node_result, sizeof(RelFileNode)), 0, "RelFileNode is different from expected");
+    else
+        TEST_RESULT_PTR(node_result, NULL, "node_result is not empty");
+    memFree(record);
+}
+
+/***********************************************************************************************************************************
+Test Run
+***********************************************************************************************************************************/
+static void
+testRun(void)
+{
+    FUNCTION_HARNESS_VOID();
+    Buffer *wal;
+    IoFilter *filter;
+    Buffer *result;
+    XLogRecord *record;
+
+    if (testBegin("read begin of the record from prev file"))
```

Comment:
Is async get tested? Didn't find any mentions of `cfgOptArchiveAsync`. Can you add tests for async get?